src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/machines/BigStorageUnit.java

```diff
@@ -135,6 +138,20 @@ public void setStoredItemType(ItemStack stored) {
         updateSignItemLines();
     }
 
+    @Nullable
+    private ItemStack getStoredItemDisplay() {
+        if (this.stored != null) {
+            ItemStack displayItemStack = stored.clone();
+            ItemMeta meta = displayItemStack.getItemMeta();
+            List<String> lore = meta != null && meta.getLore() != null ? meta.getLore() : new ArrayList<>();
+            lore.add(ChatColor.GRAY + "Stored Item");
+            meta.setLore(lore);
+            displayItemStack.setItemMeta(meta);
+            return displayItemStack;
+        }
+        return null;
+    }
```

Comment:
You should really be caching this value.

Reply:
Hopefully I took your meaning in 8d93862?

Reply:
You are calling `meta.getLore()` here twice O.o
That's a very expensive operation as it reads the NBT data and converts it to TextComponents, then String. It takes relatively long and shouldn't be ran this redundantly, cache it.